sktime/forecasting/compose/_ensemble.py

```diff
@@ -108,20 +109,43 @@ def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
         y_pred : pd.Series
             Aggregated predictions.
         """
+        aggfunc = self._check_aggfunc()
         if return_pred_int:
             raise NotImplementedError()
 
         y_pred = pd.concat(self._predict_forecasters(fh, X), axis=1)
+        return _aggregate(y=y_pred, aggfunc=aggfunc)
 
-        valid_aggfuncs = ("median", "mean", "min", "max")
-        if self.aggfunc not in valid_aggfuncs:
-            raise ValueError(f"Invalid `aggfunc`. Please use one of {valid_aggfuncs}")
-
-        if self.aggfunc == "median":
-            return y_pred.median(axis=1)
-        elif self.aggfunc == "min":
-            return y_pred.min(axis=1)
-        elif self.aggfunc == "max":
-            return y_pred.max(axis=1)
-        else:
-            return y_pred.mean(axis=1)
+    def _check_aggfunc(self):
+        valid_aggfuncs = {
+            "mean": np.mean,
+            "median": np.median,
+            "average": np.average,
+            "min": np.min,
+            "max": np.max,
+        }
+        if self.aggfunc not in valid_aggfuncs.keys():
+            raise ValueError("Aggregation function %s not recognized." % self.aggfunc)
+        return valid_aggfuncs[self.aggfunc]
+
+
+def _aggregate(y, aggfunc, X=None):
```

Comment:
```suggestion
def _aggregate(y, aggfunc):
```
unless there are aggregations that depend on X?